src/app/Login/page.tsx

```diff
@@ -2,40 +2,88 @@
 import React, { useState } from 'react';
 import ActiveButton from '@/components/Active-Button/ActiveButton';
 import InactiveButton from '@/components/Inactive-Button/InactiveButton';
-import InputField from '@/components/Input-textbox/textbox';
+import InputField from '@/components/textbox/textbox';
 import styles from './page.module.css';
 import { emailRegex } from '@/utils/constants';
 
 const LoginPage: React.FC = () => {
   // State for email and password
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [emailerror, EsetError] = useState(' ');
+  const [passerror, PsetError] = useState(' ');
+  const [shakeEmail, setShakeEmail] = useState(false);
+  const [shakePassword, setShakePassword] = useState(false);
 
-  // Handlers for input changes
   const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const value = e.target.value;
     setEmail(value);
+    if (!value) {
+      EsetError('Email cannot be empty.');
+    } else if (!emailRegex.test(value)) {
+      EsetError('Invalid email address.');
+    } else {
+      EsetError(''); // Clear error if valid
+    }
   };
 
   const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const value = e.target.value;
     setPassword(value);
+    if (!value) {
+      PsetError('Password cannot be empty.');
+    } else {
+      PsetError(''); // Clear error if valid
+    }
   };
 
   const handleSubmit = (e: React.FormEvent) => {
     e.preventDefault();
+    // Reset error messages
+    EsetError('');
+    PsetError('');
+    setShakeEmail(false);
+    setShakePassword(false);
 
-    // Check validation on submit
-    emailRegex.test(email);
+    let valid = true; // To track overall validity
+
+    if (!email) {
+      EsetError('Email cannot be empty.');
+      setShakeEmail(true); // Trigger shake for email
+      valid = false;
+    } else if (!emailRegex.test(email)) {
+      EsetError('Invalid email address.');
+      setShakeEmail(true); // Trigger shake for email
+      valid = false;
+    }
+
+    if (!password) {
+      PsetError('Password cannot be empty.');
+      setShakePassword(true); // Trigger shake for password
+
+      valid = false;
+    }
+    if (!valid) {
+      // Reset shake state after animation duration
+      setTimeout(() => {
+        setShakeEmail(false);
+        setShakePassword(false);
+      }, 100); // Match this duration with your CSS animation duration
+      return;
+    }
   };
 
   return (
     <main className={styles.loginPageContainer}>
+      <meta
+        name="viewport"
+        content="width=device-width, height=device-height, initial-scale=1"
+      ></meta>
```

Comment:
delete, the meta tag should not be defined here, metadata is only defined in `layout.tsx`. your story does not need you to work on the meta tag